### ISSUE TITLE
BadImageFormatException on .NET Core 3.1.202 and 3.1.300

### ISSUE THREAD

**Issue by poita66** [reinforced/Reinforced.Typings#174]
Hi Pavel, thank you so much for this tool and I hope everything's going well with your job!

I'm having trouble building my app with .NET Core 3.1.300 and RT 1.5.7 because of this error:

```
Reinforced.Typings : Build warning RT0099: Assembly Microsoft.AspNetCore.Mvc.Core from /home/petey/.nuget/packages/microsoft.aspnetcore.app.ref/3.1.2/ref/netcoreapp3.1/Microsoft.AspNetCore.Mvc.Core.dll was not loaded: System.BadImageFormatException: Could not load file or assembly 'Microsoft.AspNetCore.Mvc.Core, Version=3.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60, processorArchitecture=MSIL'. Reference assemblies should not be loaded for execution.  They can only be loaded in the Reflection-only loader context. (0x80131058)
Reinforced.Typings : Unexpected error RT0999: Could not load file or assembly 'Microsoft.AspNetCore.Mvc.Core, Version=3.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'. The system cannot find the file specified.
```

In our build pipeline I'm running .NET Core 3.1.201 and that seems to work, so I rolled back my development SDK to the same version and it works, so it's not blocking me for now, but others will run into this issue.

So in summary: RT doesn't seem to work on any version newer than .NET Core 3.1.201. I've tried 3.1.202 and 3.1.300 and they both fail.

Thanks again!

**Comment by zivillian** [comment]
Since #173 is already closed, I'll comment here.

I have the same problem with 3.1.301 - the ref assemblies have the version 3.1.3 - the real assemblies are version 3.1.5. I found the [FX_DEPS_FILE property](https://source.dot.net/#Microsoft.Extensions.DependencyModel/DependencyContextPaths.cs,37) which is used by the [AppBaseCompilationAssemblyResolver](https://source.dot.net/#Microsoft.Extensions.DependencyModel/Resolution/AppBaseCompilationAssemblyResolver.cs,11).

I've added a build warning to rtcli which returns `C:\Program Files\dotnet\shared\Microsoft.NETCore.App\3.1.5\Microsoft.NETCore.App.deps.json` when invoked during build.

**Comment by pavel-b-novikov** [comment]
Must be fixed by @zivillian 

Thanks a lot
